src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java

```diff
@@ -129,39 +121,56 @@ public StateFlow build() {
                 .transitionTo(PAST_CLAIM_DETAILS_NOTIFICATION_DEADLINE_AWAITING_CAMUNDA)
                     .onlyIf(pastClaimDetailsNotificationDeadline)
             .state(CLAIM_DETAILS_NOTIFIED)
-                .transitionTo(CLAIM_DETAILS_NOTIFIED_TIME_EXTENSION).onlyIf(claimDetailsNotifiedTimeExtension)
+                .transitionTo(CLAIM_DETAILS_NOTIFIED_TIME_EXTENSION)
+                    .onlyIf(respondent1TimeExtension.and(not(notificationAcknowledged)))
                 .transitionTo(NOTIFICATION_ACKNOWLEDGED).onlyIf(notificationAcknowledged)
-                .transitionTo(FULL_DEFENCE).onlyIf(fullDefenceAfterNotifyDetails)
-                .transitionTo(FULL_ADMISSION).onlyIf(fullAdmissionAfterNotifyDetails)
-                .transitionTo(PART_ADMISSION).onlyIf(partAdmissionAfterNotifyDetails)
-                .transitionTo(COUNTER_CLAIM).onlyIf(counterClaimAfterNotifyDetails)
+                .transitionTo(FULL_DEFENCE)
+                    .onlyIf(fullDefence.and(not(notificationAcknowledged.or(respondent1TimeExtension))))
+                .transitionTo(FULL_ADMISSION)
+                    .onlyIf(fullAdmission.and(not(notificationAcknowledged.or(respondent1TimeExtension))))
+                .transitionTo(PART_ADMISSION)
+                    .onlyIf(partAdmission.and(not(notificationAcknowledged.or(respondent1TimeExtension))))
+                .transitionTo(COUNTER_CLAIM)
+                    .onlyIf(counterClaim.and(not(notificationAcknowledged.or(respondent1TimeExtension))))
                 .transitionTo(TAKEN_OFFLINE_BY_STAFF).onlyIf(takenOfflineByStaffAfterClaimDetailsNotified)
                 .transitionTo(PAST_CLAIM_DISMISSED_DEADLINE_AWAITING_CAMUNDA).onlyIf(caseDismissedAfterDetailNotified)
             .state(CLAIM_DETAILS_NOTIFIED_TIME_EXTENSION)
-                .transitionTo(NOTIFICATION_ACKNOWLEDGED).onlyIf(notificationAcknowledgedTimeExtension)
-                .transitionTo(FULL_DEFENCE).onlyIf(fullDefence)
-                .transitionTo(FULL_ADMISSION).onlyIf(fullAdmission)
-                .transitionTo(PART_ADMISSION).onlyIf(partAdmission)
-                .transitionTo(COUNTER_CLAIM).onlyIf(counterClaim)
+                .transitionTo(NOTIFICATION_ACKNOWLEDGED).onlyIf(notificationAcknowledged)
+                .transitionTo(FULL_DEFENCE)
+                    .onlyIf(respondent1TimeExtension.and(not(notificationAcknowledged)).and(fullDefence))
```

Comment:
There is no need to re[eat the predicate `respondent1TimeExtension` since the only way to reach flow state `CLAIM_DETAILS_NOTIFIED_TIME_EXTENSION` is for this predicate to be `TRUE`. We can therefore simplify as below: 
![image](https://user-images.githubusercontent.com/138029/127668044-77667ca4-660a-4f5d-8338-54b325b7a44c.png)